x-pack/plugins/alerting/server/usage/lib/get_telemetry_from_event_log.ts

```diff
@@ -0,0 +1,620 @@
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+              percentile_scheduled_actions: generatedActionsPercentilesAgg,
+              percentile_alerts: alertsPercentilesAgg,
+            },
+          },
+        },
+      },
+    };
+
+    logger.debug(`query for getExecutionsPerDayCount - ${JSON.stringify(query)}`);
+    const results = await esClient.search(query);
+
+    logger.debug(`results for getExecutionsPerDayCount query - ${JSON.stringify(results)}`);
+
+    const totalRuleExecutions =
+      typeof results.hits.total === 'number' ? results.hits.total : results.hits.total?.value;
+
+    const aggregations = results.aggregations as {
+      by_rule_type_id: AggregationsTermsAggregateBase<GetExecutionCountsAggregationBucket>;
+      execution_failures: IGetExecutionFailures;
+      percentile_scheduled_actions: AggregationsPercentilesAggregateBase;
+      percentile_alerts: AggregationsPercentilesAggregateBase;
+      avg_execution_time: AggregationsSingleMetricAggregateBase;
+      avg_es_search_duration: AggregationsSingleMetricAggregateBase;
+      avg_total_search_duration: AggregationsSingleMetricAggregateBase;
+    };
+
+    const aggregationsByRuleTypeId: AggregationsBuckets<GetExecutionCountsAggregationBucket> =
+      aggregations.by_rule_type_id.buckets as GetExecutionCountsAggregationBucket[];
+
+    return {
+      ...parseRuleTypeBucket(aggregationsByRuleTypeId),
+      ...parseExecutionFailureByRuleType(aggregationsByRuleTypeId),
+      ...parseExecutionCountAggregationResults(aggregations),
+      countTotalRuleExecutions: totalRuleExecutions ?? 0,
+    };
+  } catch (err) {
+    logger.warn(
+      `Error executing alerting telemetry task: getExecutionsPerDayCount - ${JSON.stringify(err)}`
+    );
+    return {
+      countTotalRuleExecutions: 0,
+      countRuleExecutionsByType: {},
+      countTotalFailedExecutions: 0,
+      countFailedExecutionsByReason: {},
+      countFailedExecutionsByReasonByType: {},
+      avgExecutionTime: 0,
+      avgExecutionTimeByType: {},
+      avgEsSearchDuration: 0,
+      avgEsSearchDurationByType: {},
+      avgTotalSearchDuration: 0,
+      avgTotalSearchDurationByType: {},
+      generatedActionsPercentiles: {},
+      generatedActionsPercentilesByType: {},
+      alertsPercentiles: {},
+      alertsPercentilesByType: {},
+    };
+  }
+}
+
+export async function getExecutionTimeoutsPerDayCount({
+  esClient,
+  eventLogIndex,
+  logger,
+}: Opts): Promise<GetExecutionTimeoutsPerDayCountResults> {
+  try {
+    const query = {
+      index: eventLogIndex,
+      size: 0,
+      body: {
+        query: {
+          bool: {
+            filter: {
+              bool: {
+                must: [
+                  {
+                    term: { 'event.action': 'execute-timeout' },
+                  },
+                  {
+                    term: { 'event.provider': 'alerting' },
```

Comment:
Feels like some of this could be extracted - action/provide/time range anyway (action parameterized for the cases we're using), cut down on some boilerplate/size.

Reply:
Updated here: https://github.com/elastic/kibana/pull/134769/commits/a444b09a8fedc806e29ac209ba0c0725889c8db4